.github/workflows/guard-main-origin.yml

```
@@ -0,0 +1,18 @@
+name: Guard-Main-Origin
+
+on:
+  pull_request:
+    branches: [main]            # PRs whose *target* is main only
+
+jobs:
+  ensure-origin-dev:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Fail if PR head is not dev
+        run: |
```

actionlint (1.7.7)

"github.head_ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions for more details

(expression)